linode/placementgroupassignment/framework_resource.go

```diff
@@ -0,0 +1,284 @@
+
+	pg, err := client.GetPlacementGroup(ctx, pgID)
+	if err != nil {
+		if linodego.IsNotFound(err) {
+			resp.Diagnostics.AddWarning(
+				"Placement Group No Longer Exists",
+				fmt.Sprintf(
+					"Removing Placement Group assignment %s from state because the "+
+						"target Placement Group no longer exists",
+					state.ID.String(),
+				),
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Failed to get Placement Group",
+			err.Error(),
+		)
+		return
+	}
+
+	// If the Linode doesn't exist in the PG due to external modification,
+	// we should mark this assignment for recreation
+	if !pgHasID(*pg, linodeID) {
+		resp.Diagnostics.AddWarning(
+			"Marking Assignment for Recreation",
+			fmt.Sprintf(
+				"Linode (%d) is no longer assigned to target Placement Group (%d).",
+				pg.ID,
+				linodeID,
+			),
+		)
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	state.Flatten(*pg, linodeID, false, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	tflog.Debug(ctx, "Update linode_placement_group_assignment")
+	resp.Diagnostics.AddWarning(
+		"Unintended Calling to Update Function",
+		"The Update function of 'linode_placement_group_assignment' should never be "+
+			"invoked by design. This function has been redundantly implemented "+
+			"for improved reliability. Please consider reporting this as a bug "+
+			"to the provider developers.",
+	)
+
+	var state, plan PGAssignmentModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	plan.CopyFrom(state, true)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *Resource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
+	tflog.Debug(ctx, "Delete linode_placement_group_assignment")
+	var state PGAssignmentModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	client := r.Meta.Client
+
+	pgID, linodeID := state.GetIDComponents(&resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, "client.UnassignPlacementGroupLinodes(...)", map[string]any{
+		"placement_group_id": pgID,
+		"linode_id":          linodeID,
+	})
+
+	_, err := client.UnassignPlacementGroupLinodes(ctx, pgID, linodego.PlacementGroupUnAssignOptions{
+		Linodes: []int{linodeID},
+	})
+	if err != nil {
+		if linodego.IsNotFound(err) {
+			resp.Diagnostics.AddWarning(
+				fmt.Sprintf(
+					"Attempted to delete PG assignment to Linode %d but the resource was not found",
+					linodeID,
+				),
+				err.Error(),
+			)
+		} else if lErr, ok := err.(*linodego.Error); ok &&
+			lErr.Code == 400 &&
+			strings.Contains(lErr.Message, "does not belong to Placement Group") {
```

Comment:
Checking the error message here is definitely not ideal but I couldn't think of another way to differentiate between an invalid request body and an already-unassigned Linode.